nilearn/reporting/glm_reporter.py

```diff
@@ -222,744 +167,45 @@ def make_glm_report(
-            precision=2,
-            header=True,
-            sparsify=False,
-        )
-
-    contrasts = coerce_to_dict(contrasts)
-
-    # If some contrasts are passed
-    # we do not rely on filenames stored in the model.
-    output = None
-    if contrasts is None:
-        output = model._reporting_data.get("filenames", None)
-        if output is not None and output["use_absolute_path"]:
-            output = _turn_into_full_path(output, output["dir"])
-
-    design_matrices = None
-    mask_plot = None
-    mask_info = {"n_elements": 0, "coverage": "0"}
-    results = None
-
-    if not model.__sklearn_is_fitted__():
-        warning_messages.append(UNFITTED_MSG)
-
-    else:
-        design_matrices = (
-            [model.design_matrix_]
-            if model.__str__() == "Second Level Model"
-            else model.design_matrices_
-        )
-
-        if bg_img == "MNI152TEMPLATE":
-            bg_img = MNI152TEMPLATE if model._is_volume_glm() else None
-        if (
-            not model._is_volume_glm()
-            and bg_img
-            and not isinstance(bg_img, SurfaceImage)
-        ):
-            raise TypeError(
-                "'bg_img' must a SurfaceImage instance. "
-                f"Got {bg_img.__class__.__name__}"
-            )
-
-        mask_plot = _mask_to_plot(model, bg_img, cut_coords)
-
-        # We try to rely on the content of glm object only
-        # by reading images from disk rarther than recomputing them
-        mask_info = {
-            k: v
-            for k, v in model.masker_._report_content.items()
-            if k in ["n_elements", "coverage"]
-        }
-        if "coverage" in mask_info:
-            mask_info["coverage"] = f"{mask_info['coverage']:0.1f}"
-
-        statistical_maps = {}
-        if model._is_volume_glm() and output is not None:
-            try:
-                statistical_maps = {
-                    contrast_name: output["dir"]
-                    / output["statistical_maps"][contrast_name]["z_score"]
-                    for contrast_name in output["statistical_maps"]
-                }
-            except KeyError:  # pragma: no cover
-                if contrasts is not None:
-                    statistical_maps = model._make_stat_maps(
-                        contrasts,
-                        output_type="z_score",
-                        first_level_contrast=first_level_contrast,
-                    )
-        elif contrasts is not None:
-            statistical_maps = model._make_stat_maps(
-                contrasts,
-                output_type="z_score",
-                first_level_contrast=first_level_contrast,
-            )
-
-        logger.log(
-            "Generating contrast-level figures...", verbose=model.verbose
-        )
-        results = _make_stat_maps_contrast_clusters(
-            stat_img=statistical_maps,
-            threshold_orig=threshold,
+    warnings.warn(
+        category=DeprecationWarning,
```

Comment:
```suggestion
        category=FutureWarning,
```

We need to use a FutureWarning otheriwise it may not show for users unless they run in development mode: https://docs.python.org/3/library/devmode.html#devmode

Reply:
Done.